lte/gateway/c/session_manager/SessionState.cpp

```diff
@@ -550,6 +549,10 @@ void SessionState::add_rule_usage(
       rule_id, rule_version, used_tx, used_rx, dropped_tx, dropped_rx,
       session_uc);
   if (!rule_delta) {
+    if (strcmp(rule_id.c_str(), DROP_ALL_RULE) == 0) {
```

Comment:
Should we compare first, and then skip `get_rule_delta` if it's a drop rule?

Reply:
Also this being c++, you can simply add DROP_ALL_RULE as a c++ string and compare `rule_id == DROP_ALL_RULE`, this is much safer than using strcmp.